SU-HSDO/suhumsci
# STN-329: Traditional Utility Styles

# READY FOR REVIEW

## Summary
This PR adds traditional styles for the following:
- hb-categories
- hb-pill/hb-pill-list/hb-pill-link-list
- hs-font-splash
- hs-button
- hs-button--big
- hs-secondary-button

This PR also includes a significant refactor to the link/button styles to remove the use of `!important` and there is a fix for buttons in the footer of the colorful theme that was caused by the use of `!important`. This refactor includes:

- moving all button styles to `components/_buttons.scss`
- creating a `button-base` mixin
- reordering CSS to include button styles at the end of the list of component partials
- increasing specificity for button styles when classes are included on a parent element (`hb-button a:not([class])`)

**Designs:** https://www.figma.com/file/cySH8EVRf71kTj866OQmfM/%F0%9F%8E%A8--Traditional-Theme---Team-Library?node-id=0%3A1

**Browser tested in the following browsers:**
- Chrome
- Firefox
- Safari
- Edge
- IE 11

## Need Review By (Date)
asap

## Urgency
medium

## Steps to Test
1. Run `npm test`
2. Run `npm start` to compile CSS
3. Test that categories, splash font, and pill utilities match the designs in traditional.
4. Test that the buttons for exposed filters match the designs (_note: the font size for these buttons should be 18px instead of the standard 16px_). Check hover and focus states.
5. Test that all examples of inline buttons on the [QA page](http://economics.suhumsci.loc/qa/qa-page) match the designs. Check hover and focus states.
6. Test each of the button styles in the local footer. Check hover and focus states.
7. Install the colorful theme and test button styles in the local footer. Check hover and focus states and check both footer variants.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
